llvm/lib/IR/Core.cpp

```diff
@@ -178,6 +179,18 @@ LLVMTypeRef LLVMGetTypeAttributeValue(LLVMAttributeRef A) {
   return wrap(Attr.getValueAsType());
 }
 
+LLVMAttributeRef LLVMCreateConstantRangeAttribute(
+    LLVMContextRef C, unsigned KindID, unsigned NumBits, unsigned LowerNumWords,
+    const uint64_t LowerWords[], unsigned UpperNumWords,
+    const uint64_t UpperWords[]) {
```

Comment:
Another option we have would be to say that LowerWords and UpperWords must be `ceil_div(NumBits, 64)` long and not pass separate LowerNumWords and UpperNumWords arguments. Any thoughts on that?

Reply:
hmm maybe, in one way it is nice to be explicit but I guess that most users do not care and will always create an array of two elements

Reply:
maybe redundant to have two numWords assume that same array size will be used for both. 
not many other functions that have an array only LLVMConstIntOfArbitraryPrecision before this and it have the numwords do not know if we want to stick to a pattern.

Reply:
pushed up a commit with only one numWords can change if you feel like it is better to remove it also

Reply:
Not entirely sure on this, but I think it would be better to drop it. In bitcode, we encode APInts with leading zeros with less words than their bitwidth would ordinarily use, and have to separately consider BitWidth and NumWords because of that. But I can't really think of a good reason why frontend callers of this API would need this kind of micro-optimization.

Reply:
ok remove it.
 I assume that ceil_div is a fictive function or I was not able to find it.

Reply:
It's called divideCeil :)

Reply:
did not think of swapping the names but in hindsight feels like a logical order :)